File: leet_code/Beginer/soma_dois_numeros_com_target/soma_dois_numeros_com_target.py
import time


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        self.target = target
        dict_result = {}
        list_indexes = []

        for index, num in enumerate(nums):
            complemento = self.target - num
            if complemento not in dict_result.keys():
                dict_result[num] = index
            else:
                list_indexes.append(dict_result[complemento])
                list_indexes.append(index)
        if len(list_indexes) == 0:
            raise Exception("Não há solução para o target")

        return list_indexes

File: leet_code/Beginer/soma_dois_numeros_com_target/soma_dois_numeros_com_target.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let map_indexes = {};
    for (let i = 0; i < nums.length; i++) {
        let complemento = target - nums[i];
        if (complemento in map_indexes) {

            return [map_indexes[complemento], i];
        }

        map_indexes[nums[i]] = i;
    }

    map_indexes = null;
    if (map_indexes == null) {
        throw new Error("Não há solução para o target");
    }
    return [];
};

